add new iwconfig parser

vrince opened the pull request:
* [x] introduce new `iwconfig` parser
* [x] add new unit test + test data (simple case)
* [x] implement `_process` (convert int float and all)
* [x] document schema
* [x] add example in docsrting

Closes #329

kellyjonbrazil commented:
Sweet - thank you for the contribution! I’ll take a closer look tomorrow. Much appreciated!

kellyjonbrazil commented on jc/parsers/iwconfig.py:
```diff
@@ -0,0 +1,176 @@
+    raw_output: List[Dict] = []
+
+    # for backwards compatibility, preset all fields to None
+    wireless_extension_obj: Dict = {
```
You can go ahead and make this a blank dictionary. This was just done for backwards compatibility with older versions of the `ifconfig` parser after I rewrote it.

kellyjonbrazil commented on tests/test_iwconfig.py:
```diff
@@ -0,0 +1,42 @@
+        """
+        self.assertEqual(jc.parsers.iwconfig.parse('', quiet=True), [])
+
+    def test_iwconfig_ubuntu_20_04(self):
```
Might add `_raw` to the function name so it doesn't override the other test.

vrince commented:
@kellyjonbrazil I moved the test fixture to `generic` no clue how generic `iwconfig` is ... but it felt wrong to put it in ubuntu 20.10 arbitrarily. I also added the parser to the parser list so `iwconfig 2> /dev/null | jc --iwconfig -p` is actually working ;)

kellyjonbrazil commented:
All good - we can put OS-specific test fixtures in their folders if we run into any OS/distro-specific issues later. Looks very clean!
